TerraformPluginDotNet/Schemas/SchemaBuilder.cs

```diff
@@ -29,16 +30,29 @@ public Schema BuildSchema(Type type)
         var block = new Schema.Types.Block();
         foreach (var property in properties)
         {
-            var key = property.GetCustomAttribute<MessagePack.KeyAttribute>();
+            var key = property.GetCustomAttribute<KeyAttribute>();
+            if (key == null)
+            {
+                _logger.LogWarning($"{nameof(KeyAttribute)} attribute required for property {property.Name} in {type.Name}");
+                continue;
+            }
+
             var description = property.GetCustomAttribute<DescriptionAttribute>();
             var required = IsRequiredAttribute(property);
             var computed = property.GetCustomAttribute<ComputedAttribute>() != null;
 
+            var terraformType = GetTerraformType(property.PropertyType);
+            if (terraformType == null)
+            {
+                _logger.LogWarning($"Unable to convert the type {property.PropertyType.FullName} of property {property.Name} in {type.Name} to Terraform type.");
```

Comment:
I think it would be best to keep this as an exception as opposed to a warning, as it was before. Otherwise it allows using unsupported types as properties in the model and is more likely to lead to bugs in providers.

Similar for the missing `[Key]` attribute on line 36.

Reply:
Ok that is fine. A missing `[Key]` resulted in a `NullReferenceException` before, so I'll implement it as `InvalidOperationException` to be somewhat more helpful.